shanoir-ng-datasets/src/main/java/org/shanoir/ng/dataset/modality/MrDataset.java

```diff
@@ -76,7 +79,49 @@ public class MrDataset extends Dataset {
 	/** Store temporarily the last image acquisition time until all images are processed */
 	@Transient
 	private LocalDateTime lastImageAcquisitionTime;
-	
+
+	public MrDataset() {}
+
+	public MrDataset(Dataset d, MrProtocol mrpro) {
+		super(d);
```

Comment:
You could do a 
MrDataset mrDataset = (MrDataset) d;
to simplify the later code !